src/wallet/rpcdump.cpp

```diff
@@ -729,12 +729,62 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     for (std::vector<std::pair<int64_t, CKeyID> >::const_iterator it = vKeyBirth.begin(); it != vKeyBirth.end(); it++) {
         const CKeyID &keyid = it->second;
         std::string strTime = EncodeDumpTime(it->first);
-        std::string strAddr = EncodeDestination(keyid);
+        std::string strAddr;
+        std::string strLabel;
         CKey key;
         if (pwallet->GetKey(keyid, key)) {
+            bool fLabelFound = false;
+            if (!key.IsCompressed()) {
```

Comment:
Would make sense to encaptulate this (go from key/keyid to label,addr) out to a function.

Reply:
Thanks for having a look.  I'm not sure what I should be passing to that function though, seems that I would have to pass a pointer to the wallet itself along with the keyid in order to know whether a label exists for an address type for that keyid.
I might be missing something obvious though.